File: querydsl-libraries/querydsl-jpa/src/test/java/com/querydsl/jpa/domain/MoneyConverter.java
package com.querydsl.jpa.domain;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import java.math.BigDecimal;

@Converter(autoApply = true)
public class MoneyConverter implements AttributeConverter<Money, BigDecimal> {

  @Override
  public BigDecimal convertToDatabaseColumn(Money attribute) {
    return attribute != null ? attribute.getValue() : null;
  }

  @Override
  public Money convertToEntityAttribute(BigDecimal dbData) {
    return dbData != null ? new Money(dbData) : null;
  }
}

File: querydsl-examples/querydsl-example-ksp-codegen/src/main/kotlin/com/querydsl/example/ksp/MoneyConverter.kt
package com.querydsl.example.ksp

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.math.BigDecimal

@Converter
class MoneyConverter : AttributeConverter<Money, BigDecimal> {
    
    override fun convertToDatabaseColumn(attribute: Money?): BigDecimal? {
        return attribute?.value
    }
    
    override fun convertToEntityAttribute(dbData: BigDecimal?): Money? {
        return dbData?.let { Money(it) }
    }
}

